addons/l10n_ec/migrations/3.4/post-migration.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, SUPERUSER_ID
+
+def update_withhold_type(env):
+    # reclasifies withhold taxes into independent tax groups for sales and purchases
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_vat_sale') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_vat') and type_tax_use='sale')
+    ''')
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_vat_purchase') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_vat') and type_tax_use='purchase')
+    ''')
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_income_sale') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_income_tax') and type_tax_use='sale')
+    ''')
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_income_purchase') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_income_tax') and type_tax_use='purchase')
+    ''')
+
+def update_type_tax_use(env):
+    # sets type_tax_use = none for withholding taxes
+    env.cr.execute('''
+        update account_tax
+        set type_tax_use = 'none'
+        where tax_group_id in (select id from account_tax_group where l10n_ec_type in ('withhold_income_purchase','withhold_vat_purchase','withhold_income_sale','withhold_vat_sale'))
+    ''')
+
+def unlink_old_withhold_group(env):
+    deprecated_withhold_vat_group = env.ref('l10n_ec.tax_group_withhold_vat', False)
+    deprecated_withhold_profit_group = env.ref('l10n_ec.tax_group_withhold_income', False)
+    deprecated_withhold_vat_group and deprecated_withhold_vat_group.unlink()
+    deprecated_withhold_profit_group and deprecated_withhold_profit_group.unlink()
+
+def update_vat_withhold_base_percent(env):
+    # For vat withhold taxes, replace factor_percent=12% with factor_percent=100%
+    all_companies = env['res.company'].search([])
+    ecuadorian_companies = all_companies.filtered(lambda r: r.country_code == 'EC')
+    ecuadorian_taxes = env['account.tax'].search([('company_id', 'in', ecuadorian_companies.ids)])
+    taxes_to_fix = ecuadorian_taxes.filtered(lambda x: x.tax_group_id.l10n_ec_type in ['withhold_vat_sale', 'withhold_vat_purchase'])
```

Comment:
This could be incorporated in the query.

Reply:
I am not sure we need to do it in pre instead of post.  Just that it might be stabler for some cases.

Reply:
@vbe-odoo @jco-odoo this comment is ready and should just be closed, that part of the script was removed as was not strictly needed, so no need to keep discussing it.